### ISSUE TITLE
Neptune to_property_graph errors if the DataFrame is empty

### ISSUE THREAD

**Issue by brianok-aws** [aws/aws-sdk-pandas#2517]
### Describe the bug

When writing a dataframe with zero records to Neptune using to_property_graph, an exception is thrown "Error during serialization: Serializer for type com.amazon.neptune.tinkerpop.structure.NeptuneGraph$NeptuneGraphTraversalSource not found"

### How to Reproduce

```
*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*
```
Create a data frame with zero records and try to write it to Neptune using to_property_graph(). 

### Expected behavior

Nothing will be written but an exception will not be thrown.  

### OS

Amazon Linux 2 (via Jupyter Notebook on Sagemaker)

### Python version

3.10

### AWS SDK for pandas version

3.4.1

### Additional context

Related to https://github.com/aws/aws-sdk-pandas/issues/2494

**Comment by kukushking** [comment]
Hi @brianok-aws we can certainly skip the operation and return if there are no rows in the property graph df. Do you foresee any cases where user/client code would like to know they are trying to insert something that does not make sense?

**Comment by jaidisido** [comment]
I would be in favour of raising an [EmptyDataFrame](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/exceptions.py#L60) Exception rather than failing silently. Users should realise that their df is empty and they catch the exception on their own if they wish to ignore it

**Comment by brianok-aws** [comment]
I agree.  Any clear exception would be good for this.  The customer can always catch and eat it if they have zero-record updates as part of their workflow (or check the count prior to calling).  The current exception is essentially the equivalent to NullPointerException in the Gremlin driver.

**Comment by kukushking** [comment]
Fixed in #2526.
